test/utils/k8s/service.go

```diff
@@ -0,0 +1,86 @@
+package k8s
+
+import (
+	"fmt"
+	"github.com/skupperproject/skupper/test/utils/constants"
+	apiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/retry"
+	"time"
+)
+
+func WaitForServiceToBeAvailableDefaultTimeout(ns string, kubeClient kubernetes.Interface, name string) (service *apiv1.Service, err error) {
+	return WaitForServiceToBeAvailable(ns, kubeClient, name, constants.SkupperServiceReadyPeriod)
+}
+
+func WaitForServiceToBeAvailable(ns string, kubeClient kubernetes.Interface, name string, timeout time.Duration) (service *apiv1.Service, err error) {
+
+	// Create a service informer
+	done := make(chan struct{})
+	factory := informers.NewSharedInformerFactory(kubeClient, 0)
```

Comment:
If the service already exists, when the SharedInformer is created, it will
also call the `AddFunc`, reporting the given service is added to the cached
resources list. I can provide a small reproducible snippet if needed.

Reply:
Interesting! okey, sounds good.